iris_grib/_load_convert.py

```diff
@@ -843,24 +843,34 @@ def grid_definition_template_12(section, metadata):
     # Check whether Di and Dj are as consistent as possible with that
     # interpretation - i.e. they are within 1cm.
     def check_range(v1, v2, n, d):
-        min_last = v1 + (n - 1) * (d - 1)
-        max_last = v1 + (n - 1) * (d + 1)
-        if not (min_last < v2 < max_last):
+        small = min(v1, v2)
+        large = max(v1, v2)
+        min_last = small + (n - 1) * (d - 1)
+        max_last = small + (n - 1) * (d + 1)
+        if not (min_last < large < max_last):
             raise TranslationError('Inconsistent grid definition')
```

Comment:
I think there's a good opportunity here to make the message more explanatory.
It could produce messages like "file X grid is inconsistent : abs(X2 - X1) / Dj not within Nj +/- 1."

Reply:
c2e482698ed5f115121db8530c8a7e1bb1d90848